ext/vulnsrc/redhat/redhat.go

```diff
@@ -0,0 +1,731 @@
+
+			// remember the bzip hash for this entry, so we don't re-process it again next time (if unchanged)
+			DbUpdateManifestEntrySignature(manifestEntry, datastore)
+
+		} else {
+			// this pulp manifest entry has already been processed; log and skip it
+			log.Info("Pulp manifest entry unchanged since last seen. Skipping: " + manifestEntry.BzipPath)
+		}
+	
+	}
+
+	// debug
+	log.Info(fmt.Sprintf("Updating advisory-last-checked-on date in database to: %s", time.Now().Format(AdvisoryDateFormat)))
+	// update the db ky/value entry for the advisory-last-checked-on date (current timestamp, as coarse YYYY-MM-dd format)
+	DbStoreLastAdvisoryDate(time.Now().Format(AdvisoryDateFormat), datastore)
+	
+	// update the resp flag with summary of found
+	if len(resp.Vulnerabilities) > 0 {
+		resp.FlagName = UpdaterFlag
+		resp.FlagValue = time.Now().Format(UpdaterFlagDateFormat)
+	} else {
+		log.WithField("package", "Red Hat").Debug("no update")
+	}
+
+	return resp, nil
+}
+
+// gather any non-processed pulp manifest entry advisories
+func GatherUnprocessedAdvisories(manifestEntry ManifestEntry, ovalDoc OvalV2Document, datastore database.Datastore) ([]ParsedAdvisory, error) {
+	unprocessedAdvisories := []ParsedAdvisory{}
+	
+	// get all unprocessed advisories from the oval file
+	foundAdvisories, err := ProcessAdvisoriesSinceLastDbUpdate(ovalDoc, datastore)
+	if err != nil {
+		// log error and continue
+		log.Error(err)
+		return unprocessedAdvisories, err
+	} else {
+		// append found advisories to the to-be-processed list
+		unprocessedAdvisories = append(unprocessedAdvisories, foundAdvisories...)
+	}
+
+	return unprocessedAdvisories, nil
+}
+
+func CollectVulnerabilities(advisoryDefinitions []ParsedAdvisory, ovalDoc OvalV2Document) (vulnerabilities []database.VulnerabilityWithAffected) {
+	// walk the provided set of advisory definitions
+	for _, advisoryDefinition := range advisoryDefinitions {
+		vulnerabilities = append(vulnerabilities, CollectVulnsForAdvisory(advisoryDefinition, ovalDoc)...)
+	}
+	return vulnerabilities
+}
+
+// get the set of vulns for the given advisory (full doc must also be passed, for the states/tests/objects references)
+func CollectVulnsForAdvisory(advisoryDefinition ParsedAdvisory, ovalDoc OvalV2Document) (vulnerabilities []database.VulnerabilityWithAffected) {
+	// first, check the advisory severity
+	if (IsSignificantSeverity(advisoryDefinition.Metadata.Advisory.Severity)) {
+		for _, cve := range advisoryDefinition.Metadata.Advisory.CveList {
+			packageMap := make(map[string]bool)
+			vulnerability := database.VulnerabilityWithAffected{
+				Vulnerability: database.Vulnerability{
+					Name:        cve.Value + " - " + ParseRhsaName(advisoryDefinition),
+					Link:        CveURL + cve.Value,
+					Severity:    GetSeverity(advisoryDefinition.Metadata.Advisory.Severity),
+					Description: advisoryDefinition.Metadata.Description,
+				},
+			}
+			packageList := GetPackageList(advisoryDefinition.Criteria, ovalDoc)
```

Comment:
You already have a package list stored in `advisoryDefinition.PackageList` You probably don't have to parse it again.